### ISSUE TITLE
Detection of https protocol is incorrect in Http2Curl.http2curl()

### ISSUE THREAD

**Issue by pafitchett-ks** [dzieciou/curl-logger#17]
In http2Curl() there is a block of code for detecting https protocol:

```
      if (host.endsWith(":443")) {
        inferredScheme = "https";
      } else if (request instanceof RequestWrapper) {
        if (getOriginalRequestUri(request).startsWith("https")) {
          // This is for original URL, so if during redirects we go out of HTTPs, this might be a wrong guess
          inferredScheme = "https";
        }
      }
```

This code fails if **request** is an instance of **HttpRequestWrapper** (which it will be for versions of http client >= 4.3) because **HttpRequestWrapper** does not itself derive from **RequestWrapper** and so the call to getOriginalRequestUri() is never executed.

This results in curl call output always displaying http as the protocol.

**Comment by dzieciou** [comment]
@pafitchett-ks Thank you for your report and detailed root cause analysis.

How can I reproduce the problem?

Do you have a sample REST-assured requests that causes the failure?

**Comment by pafitchett-ks** [comment]
This test, when added to *UsingWithHttpClientTest.java*, will demonstrate the problem:

```
import org.testng.Assert;
import uk.org.lidalia.slf4jtest.TestLoggerFactory;
```

```
  @Test(groups = "end-to-end-samples")
  public void testHttps() throws IOException {
    TestLoggerFactory.clearAll();
    HttpGet getRequest = new HttpGet("https://google.com");
    createHttpClient().execute(getRequest);
    Assert.assertEquals(TestLoggerFactory.getAllLoggingEvents().get(0).getMessage(),"curl 'https://google.com/' --compressed --insecure --verbose");
  }
```

**Comment by dzieciou** [comment]
@pafitchett-ks Fixed in 1.0.4-SNAPSHOT. Thanks again for reporting.
